### ISSUE TITLE
spotless-maven-plugin doesn't remove duplicate imports

### ISSUE THREAD

**Issue by mnowotnik** [diffplug/spotless#466]
Removes unused imports, but doesn't remove duplicates. It should be possible as per last comment on [Issue #28 ](https://github.com/diffplug/spotless/issues/28)
- Maven: 3.6.1
- Spotless version: spotless-maven-plugin:1.24.3
- Linux Mint 19
- Configuration:

```
            <plugin>
                <groupId>com.diffplug.spotless</groupId>
                <artifactId>spotless-maven-plugin</artifactId>
                <version>1.24.3</version>
                <executions>
                    <execution>
                        <id>spotless-check</id>
                        <phase>verify</phase>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <skip>${skipSpotless}</skip>
                    <java>
                        <removeUnusedImports/>
                    </java>
                </configuration>
            </plugin>
```
- Command: `mvn spotless:apply`

**Comment by nedtwigg** [comment]
It seems reasonable that `removeUnusedImports` ought to remove duplicates, but which one should it remove?  The first one or the last one?

Sorting the imports removes this ambiguity. If you [sort the imports](https://github.com/diffplug/spotless/tree/master/plugin-maven#applying-to-java-source) `<importOrder><order></order></importOrder>`, that will remove duplicate imports.

It seems reasonable to add some kind of duplicate removal into `removeUnusedImports`, but it's unusual to care about unused imports and not care about their order.  For `removeUnusedImports`, we are using a piece of `google-java-format`.  However, `google-java-format` normally sorts the imports, so they don't care if the `removeUnusedStep` does the removal or not.

TL;DR: this is easy to fix by sorting, and surprisingly hard to implement without sorting.

**Comment by mnowotnik** [comment]
Fair enough. However, in the case of two consecutive identical imports it could remove either one without ambiguity.

```
import a.b.c.X;
import a.b.c.X;
```

**Comment by nedtwigg** [comment]
That's true.  To implement this, someone will have to make a PR with google-java-format (not likely to be accepted, for the reasons stated above), or you'll have to write it from scratch in to our integration of google-java-format.  The problem is that your usecase "I don't want duplicates and I don't want to sort them" is rare, and implementing it is harder than just adding all the imports to a SortedSet and then dumping the result into a file.  For a maintainer, adding complexity to accommodate a rare usecase is a bad bargain.  If someone figures out a simple way to do it I'd be happy to merge, but I doubt there is a simple way.
